modules/react/common/lib/utils/components.ts

```diff
@@ -148,10 +163,15 @@ interface RefForwardingComponent<T, P = {}> {
  * return type is `Component<element, Props>` which looks like `Component<'div', Props>` which is a
  * clean interface that tells you the default element that is used.
  */
-export const createComponent = <T extends React.ElementType | undefined = undefined>(as?: T) => <
-  P,
-  SubComponents = {}
->({
+export const createComponent = <
+  E extends
+    | keyof JSX.IntrinsicElements
+    | React.ComponentType
+    | ElementComponent<any, any>
+    | undefined = undefined
```

Comment:
This was probably the source of Typescript errors when trying to use other `ElementComponent`s as an input to another component: `createComponent(Box)({...})`. While there's not always a good use-case for doing so, this removes the Typescript error.